### PR TITLE
Add support for AccountDelete

### PR DESCRIPTION
**Pull request by intelliot**
https://xrpl.org/accountdelete.html

### CODE REVIEW

File: src/common/schemas/specifications/account-delete.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "accountDelete",
+  "link": "account-delete",
+  "type": "object",
+  "properties": {
+    "destination": {
```
Comment (keefertaylor) [inline src/common/schemas/specifications/account-delete.json]: So are these properties mutually exclusive? 

Should we only be using X-Addresses?

Comment (intelliot) [reply src/common/schemas/specifications/account-delete.json]: Great catch! I’m fine with only using X-addresses here. Will update the PR later today

Comment (keefertaylor) [reply src/common/schemas/specifications/account-delete.json]: A product decision is to whether we want to support X-Addresses only. 

If you want both APIs, that's fine. But I think you then need a `oneof` in your `required` property below.

Comment (intelliot) [reply src/common/schemas/specifications/account-delete.json]: Good point, it should be fine to support both classic addresses and X-addresses. Updated.

File: src/ledger/parse/account-delete.ts [hunk 2]
```diff
@@ -11,6 +11,10 @@ export type FormattedAccountDelete = {
   // for the recipient of the deleted account's leftover XRP. NB: Ensure that the hosted recipient is
   // able to account for AccountDelete transactions; if not, your balance may not be properly credited.
   destinationTag?: number
+
+  // X-address of an account to receive any leftover XRP after deleting the sending account.
+  // Must be a funded account in the ledger, and must not be the sending account.
+  destinationXAddress: string
```
Comment (hbergren) [inline src/ledger/parse/account-delete.ts]: Does this mean you have to supply both the normal address and X Address? Ideally it's either/or:

`normal_destination_address (+ optional_tag) || x_address`

right?

Comment (intelliot) [reply src/ledger/parse/account-delete.ts]: right, it should be either/or

Comment (intelliot) [reply src/ledger/parse/account-delete.ts]: This particular file is `src/ledger/parse/account-delete.ts` which is used to parse an existing `AccountDelete` transaction from the ledger. When you retrieve an `AccountDelete` tx and it's parsed by ripple-lib, you should always get both the classic address and the X-address. So I think this code is correct

Comment (keefertaylor) [reply src/ledger/parse/account-delete.ts]: I'm not going to block this PR from going in, but that feels like a *really* weird API that isn't intuitive to use. I'd recommend just choosing one or the other (probably X-Addresses) or creating two sets of APIs. 

My rationale is that giving users too much information, or too many options, leads to confusion and greater chances of mistakes and / or bugs.

Comment (intelliot) [reply src/ledger/parse/account-delete.ts]: 👍 Let's try to move to X-address-only in ripple-lib v2.0

File: test/api/getTransaction/index.ts [hunk 3]
```diff
@@ -354,6 +354,16 @@ export default <TestSuite>{
     )
   },
 
+  'AccountDelete': async (api, address) => {
+    const hash = 'EC2AB14028DC84DE525470AB4DAAA46358B50A8662C63804BFF38244731C0CB9'
```
[the remark this answers is not in the document]
Comment (intelliot) [reply test/api/getTransaction/index.ts]: Yes

File: test/fixtures/rippled/tx/account-delete.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      "DeliveredAmount": "9995000000",
+      "TransactionIndex": 0,
+      "TransactionResult": "tesSUCCESS",
+      "delivered_amount": "9995000000"
+    },
+    "validated": true
+  },
+  "status": "success",
+  "type": "response"
+}
```
[the remark this answers is not in the document]
Comment (intelliot) [reply test/fixtures/rippled/tx/account-delete.json]: In this case, I created/submitted an AccountDelete tx on Devnet, and then retrieved it.

File: src/ledger/parse/transaction.ts [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
+import parsePaymentChannelClaim from './payment-channel-claim'
 import parsePaymentChannelCreate from './payment-channel-create'
 import parsePaymentChannelFund from './payment-channel-fund'
-import parsePaymentChannelClaim from './payment-channel-claim'
-import parseFeeUpdate from './fee-update'
-import parseAmendment from './amendment'
+import parseTrustline from './trustline'
+
+import parseAmendment from './amendment' // pseudo-transaction
+import parseFeeUpdate from './fee-update' // pseudo-transaction
```
[the remark this answers is not in the document]
Comment (intelliot) [reply src/ledger/parse/transaction.ts]: Yes, this was alphabetized to make it easier to find things, and the pseudo-transactions were separated out

**Pull request merged**